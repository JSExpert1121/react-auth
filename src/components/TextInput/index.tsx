// @material ui components
import TextField from '@material-ui/core/TextField';

// i18n
import authMsgs from 'translations/messages/page.auth';

// custom components
import withValidators from 'components/HOCs/withValidators';
import {
    isValidEmail,
    isRequired,
    isValidChar,
    isValidLengthMin,
    isValidLengthMax,
    isPasswordMatched
} from 'helper/validators';

export const EmailInput = withValidators([
    { validator: isRequired, message: authMsgs.emailRequired },
    { validator: isValidEmail, message: authMsgs.emailInvalid }
])(TextField);

export const NameInput = withValidators([
    { validator: isRequired, message: authMsgs.nameRequired }
])(TextField);

export const PassInput = withValidators([
    { validator: isRequired, message: authMsgs.passRequired },
    { validator: isValidChar, message: authMsgs.passInvalid },
])(TextField);

export const PassConfirmInput = withValidators([
    { validator: isRequired, message: authMsgs.passRequired },
    { validator: isValidChar, message: authMsgs.passInvalid },
    { validator: isValidLengthMin(8), message: authMsgs.passShort },
    { validator: isValidLengthMax(24), message: authMsgs.passLong },
    { validator: isPasswordMatched, message: authMsgs.passMismatch },
])(TextField);