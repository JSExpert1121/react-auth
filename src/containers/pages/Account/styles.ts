import { Theme, createStyles } from '@material-ui/core/styles';
import Colors from 'assets/colors';

export const styleFn = (theme: Theme) => createStyles({
    root: {
        width: '100%',
        padding: theme.spacing(3),
        maxWidth: 512,
        minWidth: 480,
    },
    form: {
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        width: '100%',
        textAlign: 'center'
    },
    logo: {
        padding: theme.spacing(2, 0)
    },
    title: {
        fontSize: '1.5rem',
        color: 'rgba(0, 0, 0, 0.8)',
        padding: theme.spacing(2)
    },
    text: {
        fontSize: '0.875rem',
        color: 'rgba(0, 0, 0, 0.7)',
        padding: theme.spacing(2)
    },
    success: {
        color: 'rgba(0, 0, 0, 0.9)',
        padding: theme.spacing(1)
    },
    error: {
        fontSize: '0.875rem',
        color: Colors.error,
        padding: theme.spacing(0, 0, 0, 2)
    },
    busy: {
        width: '100%',
    },
    submit: {
        margin: theme.spacing(2, 0),
        height: 'calc(100% - 12px)'
    },
    bottomMenu: {
        width: '100%',
        display: 'flex',
        padding: theme.spacing(2, 0)
    }
});