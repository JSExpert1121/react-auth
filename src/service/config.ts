import { oauthClient } from './base';

export const setLanguage = (lang: string) => {
    oauthClient.setLanguage(lang);
}
