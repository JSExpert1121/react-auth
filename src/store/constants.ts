
// global actions
export const GLOBAL_CHANGE_LOCALE = 'GLOBAL_CHANGE_LOCALE';
export const GLOBAL_SET_TITLE = 'GLOBAL_SET_TITLE';


// user actions
export const USER_LOGIN_SUCCESS = 'USER_LOGIN_SUCCESS';
export const USER_LOGOUT_SUCCESS = 'USER_LOGOUT_SUCCESS';
export const USER_REGISTER_SUCCESS = 'USER_REGISTER_SUCCESS';
export const USER_CHANGE_PROFILE_SUCCESS = 'USER_CHANGE_PROFILE_SUCCESS';
export const USER_GET_USER_SUCCESS = 'USER_GET_USER_SUCCESS';
export const USER_UPDATE_USER_SUCCESS = 'USER_UPDATE_USER_SUCCESS';
export const USER_GET_PROFILE_SUCCESS = 'USER_GET_PROFILE_SUCCESS';

