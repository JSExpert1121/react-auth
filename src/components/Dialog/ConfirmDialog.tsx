import React from 'react';

import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import Button from '@material-ui/core/Button';
import { makeStyles, Theme } from '@material-ui/core/styles';

// import { WrappedComponentProps, injectIntl } from 'react-intl';
// import messages from 'translations/messages/global';

const useStyles = makeStyles((theme: Theme) => ({
    relative: {
        position: 'relative',
        left: '0px',
        top: '0px',
    },
    busy: {
        position: 'absolute',
        left: 'calc(50% - 16px)',
        top: 'calc(50% - 16px)',
    },
}));

export interface IConfirmDialogProps {
    open: boolean;
    onYes: () => Promise<any> | void;
    yes?: string;
    ok?: string;
    cancel?: string;
    onCancel?: () => void;
    title?: string;
    message: string | React.ReactNode;
}

const ConfirmDialog: React.FC<IConfirmDialogProps> = props => {

    const classes = useStyles({});
    const {
        open, onYes, onCancel, title, message, yes, cancel, ok
    } = props;


    return (
        <Dialog open={open} onClose={onCancel} aria-labelledby="confirm-dialog-title">
            <DialogTitle id="confirm-dialog-title">{title}</DialogTitle>
            <DialogContent className={classes.relative}>
                <DialogContentText id="confirm-dialog-description">
                    {message}
                </DialogContentText>
            </DialogContent>
            <DialogActions>
                {onCancel && (
                    <Button id='confirm-button-cancel' onClick={onCancel} autoFocus>
                        {cancel}
                    </Button>
                )}
                <Button onClick={onYes} color="primary">
                    {onCancel ? yes : ok}
                </Button>
            </DialogActions>
        </Dialog>
    );
};

ConfirmDialog.defaultProps = {
    title: 'Confirm',
    yes: 'Yes',
    ok: 'OK',
    cancel: 'Cancel'
};

export default ConfirmDialog;
