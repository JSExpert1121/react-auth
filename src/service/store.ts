import { API_GATEWAY_ADDRESS } from 'config/service.config';
import { oauthClient } from './base';

const STORE_API_ROOT = API_GATEWAY_ADDRESS + '/api/v5/store/';

export const registerStore = (token: string, data: object) => oauthClient.authPost(STORE_API_ROOT + 'register', token, data);

export const getAllStores = (token: string) => oauthClient.authGet(STORE_API_ROOT + 'getallmystoresid', token);
