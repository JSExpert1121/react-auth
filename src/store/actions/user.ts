/**
 * @name GlobalAction
 * @description user management. sign up, sign in, reset password, ...
 * @version 0.1 (only skeleton)
 * @version 0.5 (signup)
 * @version 1.0 (signup/login/logout)
 * @todo according to spec
 * @author Darko
 */
import { User, UserProfile } from 'types/user';
import * as MemberApi from 'service/member';
import { Gender } from 'types/global';

import {
    USER_LOGIN_SUCCESS,
    USER_GET_USER_SUCCESS,
    USER_UPDATE_USER_SUCCESS,
    USER_GET_PROFILE_SUCCESS
} from '../constants';

export const loginSuccess = (token: string) => ({
    type: USER_LOGIN_SUCCESS,
    payload: token
});

export const getuserSuccess = (user: User) => ({
    type: USER_GET_USER_SUCCESS,
    payload: user
});

export const getprofileSuccess = (data: UserProfile) => ({
    type: USER_GET_PROFILE_SUCCESS,
    payload: data
});

export const updateUserSuccess = (user: User) => ({
    type: USER_UPDATE_USER_SUCCESS,
    payload: user
});

const clearLogin = dispatch => {
    dispatch(loginSuccess(undefined));
    dispatch(getuserSuccess(undefined));
    dispatch(getprofileSuccess(undefined));
}

let timeHandler = undefined;
export const refreshToken = () => async dispatch => {
    const newToken = await MemberApi.refresh();
    dispatch(loginSuccess(newToken));
    if (newToken) {
        timeHandler = setTimeout(() => dispatch(refreshToken()), 3500000);
    }
}

export const signup = (email: string, pass: string, name: string, code: string, phone: string, agrees: Array<any>) => async dispatch => {

    let data = undefined;
    data = await MemberApi.register(email, pass, name);
    // console.log(data);
    const token = await MemberApi.login(email, pass);
    dispatch(loginSuccess(token));

    data = await MemberApi.agreeTAC(token, agrees);
    data = await MemberApi.updateProfile(token, code, phone, true);

    // get user data
    data = await MemberApi.getUser(token);
    dispatch(getuserSuccess(data));

    // get user profile
    data = await MemberApi.getProfile(token);
    dispatch(getprofileSuccess(data));
}

export const login = (email: string, pass: string) => async dispatch => {
    const token = await MemberApi.login(email, pass);
    dispatch(loginSuccess(token));

    // get user data
    let data = await MemberApi.getUser(token);
    dispatch(getuserSuccess(data));

    // get user profile
    data = await MemberApi.getProfile(token);
    dispatch(getprofileSuccess(data));
}

export const logout = () => dispatch => MemberApi.logout().then(data => {
    clearLogin(dispatch);
    timeHandler && clearTimeout(timeHandler);
    timeHandler = undefined;
});

export const getUser = (token: string, id?: number) => dispatch => MemberApi.getUser(token, id).then(data => {
    dispatch(getuserSuccess(data));
    return data;
});

export const updateUser = (token: string, newMail: string, password: string, name: string, id?: number) => dispatch => MemberApi.updateUser(token, newMail, password, name, id).then(data => {
    dispatch(updateUserSuccess(data));
    return data;
});

export const deleteUser = (token: string, id?: number) => dispatch => MemberApi.deleteUser(token, id).then(data => {
    clearLogin(dispatch);
});

export const getProfile = (token: string, id?: number) => dispatch => MemberApi.getProfile(token, id).then(data => {
    dispatch(getprofileSuccess(data));
    return data;
});

export const updateProfile = (
    token: string,
    country: string,
    phone: string,
    phoneVerified: boolean,
    id?: number,
    birth?: string,
    gender?: Gender,
    photo?: string,
) => dispatch => MemberApi.updateProfile(token, country, phone, phoneVerified, id, birth, gender, photo).then(data => {
    dispatch(getprofileSuccess(data));
    return data;
});
