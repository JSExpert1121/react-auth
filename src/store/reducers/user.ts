import {
    USER_LOGIN_SUCCESS,
    USER_GET_USER_SUCCESS,
    USER_GET_PROFILE_SUCCESS
} from '../constants';
import { AnyAction } from 'redux';
import { User, UserProfile } from 'types/user';

type UserState = {
    user?: User;
    profile?: UserProfile;
    token: string;
}

const initState: UserState = {
    user: undefined,
    profile: undefined,
    token: ''
};

const userReducer: (state: UserState | undefined, action: AnyAction) => UserState = (state = initState, action) => {
    switch (action.type) {
        case USER_LOGIN_SUCCESS:
            return {
                ...state,
                token: action.payload as string
            }
        case USER_GET_USER_SUCCESS:
            return {
                ...state,
                user: action.payload
            }
        case USER_GET_PROFILE_SUCCESS:
            return {
                ...state,
                profile: action.payload
            }
        default:
            return state;
    }
}

export default userReducer;