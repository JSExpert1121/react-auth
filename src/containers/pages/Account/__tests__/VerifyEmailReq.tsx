import React from 'react';
import { MemoryRouter, withRouter } from 'react-router-dom';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16/build';

import { createMount, createShallow } from '@material-ui/core/test-utils';
import { ThemeProvider } from '@material-ui/styles';
import { IntlProvider, injectIntl } from "react-intl";
import { act } from 'react-dom/test-utils';

import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import Link from '@material-ui/core/Link';
import LinearProgress from '@material-ui/core/LinearProgress';

import { EmailInput, PassInput, NameInput } from 'components/TextInput';

import { translationMessages } from 'config/i18n.config';
import theme from 'config/theme.config';
import messages from 'translations/messages/page.account';
import { User, UserProfile } from 'types/user';
import ConfirmDialog from 'components/Dialog/ConfirmDialog';

import DefVerifyEmailReqPage, { VerifyEmailReqPage } from '../VerifyEmailReq';
// store mock
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
import { GLOBAL_SET_TITLE } from 'store/constants';
jest.mock('service/member');

configure({ adapter: new Adapter() });

const ConnectedVerifyEmailReqPage = withRouter(DefVerifyEmailReqPage);


describe('containers/pages/Account/VerifyEmailReq - request fail', () => {
    const mockMail = 'test@test.com';
    const user: User = { id: 100, name: 'testuser', username: mockMail, enabled: true };
    const priv_token = `${mockMail}__private_token`;
    const fnIntl = { formatMessage: jest.fn() };
    const setTitle = jest.fn();
    const history = { goBack: jest.fn() };

    const shallow = createShallow();
    const wrapper = shallow(
        <VerifyEmailReqPage
            user={user}
            token={priv_token}
            intl={fnIntl}
            history={history}
            setTitle={setTitle}
        />
    );

    test('UI components', () => {
        expect(wrapper.find(Typography)).toHaveLength(2);
        expect(wrapper.find(Button)).toHaveLength(1);
        expect(wrapper.find(LinearProgress)).toHaveLength(1);
    });

    test('User interaction', async done => {
        await wrapper.find(Button).props().onClick(undefined);

        setImmediate(() => {
            expect(wrapper.find(Typography)).toHaveLength(2);
            expect(wrapper.find(Button)).toHaveLength(1);
            done();
        })
    });
});

describe('containers/pages/Account/VerifyEmailReq - request success', () => {
    const mockMail = 'user1@test.com';
    const user: User = { id: 100, name: 'testuser', username: mockMail, enabled: true };
    const priv_token = `${mockMail}__private_token`;
    const fnIntl = { formatMessage: jest.fn() };
    const setTitle = jest.fn();
    const history = { goBack: jest.fn() };

    const shallow = createShallow();
    const wrapper = shallow(
        <VerifyEmailReqPage
            user={user}
            token={priv_token}
            intl={fnIntl}
            history={history}
            setTitle={setTitle}
        />
    );

    test('UI components', () => {
        expect(wrapper.find(Typography)).toHaveLength(2);
        expect(wrapper.find(Button)).toHaveLength(1);
    });

    test('User interaction - send code', async done => {
        await wrapper.find(Button).props().onClick(undefined);

        setImmediate(() => {
            wrapper.update();
            expect(wrapper.find(Typography)).toHaveLength(3);
            expect(wrapper.find(Button)).toHaveLength(0);
            expect(wrapper.find(Link)).toHaveLength(2);
            done();
        });
    });

    test('User interaction - go back', done => {
        act(() => {
            wrapper.find(Link).first().simulate('click');
        })

        setImmediate(() => {
            wrapper.update();
            expect(history.goBack).toHaveBeenCalledTimes(1);
            done();
        });
    });

    test('User interaction - resend', async done => {
        await wrapper.find(Link).last().props().onClick(undefined);

        setImmediate(() => {
            wrapper.update();
            expect(wrapper.find(Typography)).toHaveLength(3);
            expect(wrapper.find(Button)).toHaveLength(0);
            expect(wrapper.find(Link)).toHaveLength(2);
            done();
        });
    });
});


describe('containers/pages/Account/VerifyEmail - real mount', () => {

    const user: User = { id: 100, name: 'testuser', username: 'test@test.com', enabled: true };
    const priv_token = 'test@test.com__private_token';
    const profile: UserProfile = {
        cellphoneCountryCode: '82',
        cellphone: '1000000000',
        emailVerified: false,
        cellphoneVerified: true
    }

    const store = mockStore({
        user: { user, profile, token: priv_token },
    });

    const mount = createMount();
    const wrapper = mount(
        <Provider store={store}>
            <MemoryRouter>
                <IntlProvider locale='en' messages={translationMessages.en}>
                    <ThemeProvider theme={theme}>
                        {<ConnectedVerifyEmailReqPage />}
                    </ThemeProvider>
                </IntlProvider>
            </MemoryRouter>
        </Provider>
    );

    test('Snapshot test', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('setTitle', () => {
        wrapper.update();
        expect(store.getActions()).toContainEqual({
            type: GLOBAL_SET_TITLE,
            payload: 'Email Confirmation'
        });
    });
});