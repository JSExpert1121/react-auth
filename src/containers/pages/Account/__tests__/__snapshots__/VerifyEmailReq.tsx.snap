// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/pages/Account/VerifyEmail - real mount Snapshot test 1`] = `ReactWrapper {}`;
